keras_cv/layers/preprocessing/random_saturation.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class RandomSaturation(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Randomly performs the saturation adjustment on given images.
+
+    This layer will randomly increase/reduce the saturation for the input RGB
+    images. At inference time, the output will be identical to the input.
+    Call the layer with `training=True` to adjust the brightness of the input.
+
+    Note that different saturation adjustment factors will be apply to each the
+    images in the batch.
+
+    Args:
+        factor: Either a tuple of two floats or a single float. `factor` controls the
+            extent to which the image saturation is impacted. `factor=1.0` makes
+            this layer perform a no-op operation. `factor=0.0` makes the image to be
+            fully grayscale. Any value larger than 1.0 will increase the saturation
+            of the image.
+
+            Values should be between `0.0` and +inf. If a tuple is used, a `factor`
```

Comment:
what does +inf do?  Isn't 1.0 the maximum?

Reply:
The saturation is bit tricky here. Think about the `new_saturation == factor * existing_saturation`, the existing_saturation could be 0 (eg a grey scale input). The valid range of factor is [0, inf]. 

One thing we could try is to map the [0, inf] range to [0, 1], eg by f(x) = x / (1 + x), this also works when x == 1 (no saturation update). 

I also feel the factor range here is a bit different from all other factors, but this is the direct math value we give to the backend image API. If we choose to do a mapping, then we need to provide some explanation to user for how to choose the value range. 

Let me know what do u think.

Reply:
So one consideration is that in most of our layers a factor of [1.0] means you are applying the strongest possible augmentation the layer can produce, while a factor of 0.0 produces no augmentation.

Maybe we can adjust our design to map both from [0, 1] => [0, inf] and also invert the range so that it is consistent across all of our KPLs.  That way, users always know [0, 0.5] produces medium strength augmentations, and [0, 0.1] always produces weak augmentations.

Reply:
Done updated with factor value mapping. Now the input factor is [0, 1]